### ISSUE TITLE
[Bug]: ./09-PRS_EWAS: Correlation plots for PRS and non genetic methylation PCs generated

### ISSUE THREAD

**Issue by ks164** [genetics-of-dna-methylation-consortium/godmc_phase2#48]
### Contact Details

ks164@duke.edu

### Scripts

09-PRS_EWAS.sh

### What happened?

module fails when generating PRS-non-genetic PC correlation plots.

My assumption is that the error is caused by 'meth_pcs_nongenetic_untransformed.txt' being empty except for an ID column (no non-genetic methylation PCs remained for the untransformed data after running module 03d).

### R version

4.3.3 (February, 2024)

### Python version

None

### Relevant log output

```shell
Correlation plots for PRS and non genetic methylation PCs generated

Error in `purrr::map()`:
ℹ In index: 1.
Caused by error in `grobs[[i]]`:
! subscript out of bounds
Backtrace:
     ▆
  1. ├─base::print(ggarrange(plotlist = list_pc, ncol = ncol, nrow = nrow))
  2. ├─ggpubr::ggarrange(plotlist = list_pc, ncol = ncol, nrow = nrow)
  3. │ └─purrr::map(...)
  4. │   └─purrr:::map_("list", .x, .f, ..., .progress = .progress)
  5. │     ├─purrr:::with_indexed_errors(...)
  6. │     │ └─base::withCallingHandlers(...)
  7. │     ├─purrr:::call_with_cleanup(...)
  8. │     └─ggpubr (local) .f(.x[[i]], ...)
  9. │       └─cowplot::plot_grid(plotlist = plotlist, ...)
 10. │         └─cowplot::align_plots(...)
 11. └─purrr (local) `<fn>`(`<sbscOOBE>`)
 12.   └─cli::cli_abort(...)
 13.     └─rlang::abort(...)
Execution halted
```

**Comment by msolerartigas** [comment]
Hi,

Thanks for your comment!

I have updated the scripts so that they work well in that scenario of no non-genetic methylation PCs. The update is waiting to be merged into the main pipeline after that it should all work fine.

**Comment by ks164** [comment]
Thanks so much!

Karen

**Comment by NLlonga** [comment]
Hi, 

I’m encountering errors when running the 3d and 09 scripts.

`./03d-non_genetic_methylation_pcs.sh 
`

> Identified 0 out of  PCs with significant genetic component
> Error in if (length(l) == ncol(pc)) { : argument is of length zero
> Calls: main
> Execution halted

`./09-PRS_EWAS.sh 
`
> Error in fread(paste(pc_file, ".txt", sep = "")) : 
>   File '/media/mribases/DATOS1/SCRATCH/Natalia_Llonga/GoDMC2/analysis_whole_blood/processed_data/methylation_data/meth_pcs_nongenetic_untransformed.txt' does not exist or is non-readable. getwd()=='/media/mribases/DATOS1/SCRATCH/Natalia_Llonga/GoDMC2/godmc_phase2'
> Calls: fread -> stopf -> raise_condition -> signal
> Execution halted

Any help with these issues is appreciated.

Many thanks in advance, 

Natalia

**Comment by msolerartigas** [comment]
Thanks Natalia,

With regards to the 09 script, for it to work it needs a meth_pcs_nongenetic_untransformed.txt file, even if this file only has an ID column but no PCs. If 3d generates that file when the error is solved, 09 should also work.

**Comment by ZXiaopu** [comment]
Hi,

The current pipeline work like this - if there is no methylation PCs, the meth_pcs_nongenetic_untransformed.txt file will not be generated in 03d and the methylation PC adjustment step in 03e won't run.

If I generate an empty meth_pcs_nongenetic_untransformed.txt with a ID column in 03d and reimplement 03e, will it solve the issue? @msolerartigas

**Comment by msolerartigas** [comment]
Hi,

Thanks for the info!

Yes, it would solve the problem, but I guess it makes more sense that I adapt my script to look for the file and if it is not there to assume there are no non genetic methylation PCs, rather than you creating an empty file. If you agree I will go ahead and make the edits.

**Comment by ZXiaopu** [comment]
> Hi,
> 
> Thanks for the info!
> 
> Yes, it would solve the problem, but I guess it makes more sense that I adapt my script to look for the file and if it is not there to assume there are no non genetic methylation PCs, rather than you creating an empty file. If you agree I will go ahead and make the edits.

Yes it sounds good to me! Thank you

**Comment by msolerartigas** [comment]
I have now updated the scripts in my branch and I have done a pull request. Once it is merged it should solve the problems.
